### PR TITLE
all: replace deprecated ioutil references

### LINKED ISSUE #9034: cleanup: remove deprecated ioutil package
We reference the ioutil package in a few places which is deprecated. We should replace them with the new imports so VSCode stops yelling at me.

### PR DESCRIPTION
**Pull request by marwan-at-work**
This PR removes calls to ioutil library and replaces them with their new locations in the io and os packages.

Fixes #9034

### CODE REVIEW

File: tstest/iosdeps/iosdeps.go [hunk 1]
```diff
@@ -17,7 +17,6 @@ import (
 	_ "fmt"
 	_ "io"
 	_ "io/fs"
-	_ "io/ioutil"
```
Comment (marwan-at-work) [inline tstest/iosdeps/iosdeps.go:20]: is this ok to remove?

Comment (bradfitz) [reply tstest/iosdeps/iosdeps.go:20]: Yes.

**Comment by bradfitz** [comment]
In commit message, change "misc" to "all".

And add `Updates #5210`.

**Comment by marwan-at-work** [comment]
@bradfitz done, thanks!

**Review by bradfitz — APPROVED** [review]

**Pull request merged**